### PR TITLE
Add Dim method to cholesky

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

File: mat64/cholesky.go [hunk 1]
```diff
@@ -74,6 +74,11 @@ func (c *Cholesky) Det() float64 {
 	return math.Exp(c.LogDet())
 }
 
+// Dim returns the dimension of the factorized matrix.
```
Comment (kortschak) [inline mat64/cholesky.go]: Move this above Det.

**Comment by kortschak** [comment]
Is Dim the best name? I can't think of a better one, but it's very close to Dims which is one of our interface methods.

**Comment by kortschak** [comment]
LGTM

**Comment by btracey** [comment]
I'm not sure I like Dim either. Other options:
- Size
- N
- Cholesky (in the spirit of Symmetric and Triangular)

**Comment by kortschak** [comment]
Cholesky as a method name is misleading. Size works for me though I was also considering N.

**Comment by btracey** [comment]
I'll change to Size. I also thought about Dims, but I thought Cholesky even starting to look like it implements Matrix is a bad idea.

**Pull request merged**